osf/management/commands/migratedata.py

```diff
@@ -333,8 +332,10 @@ def fix_guids():
     short_missing = [x for x in missing if len(x) < 6]
     long_missing = [x for x in missing if len(x) > 5]
     assert len(short_missing) + len(long_missing) == len(missing), 'It broke'
-    short_missing_guids = MGuid.find(MQ('_id', 'in', short_missing))
-    long_missing_guids = MGuid.find(MQ('_id', 'in', long_missing))
+    # NOTE: is_orphaned will be True for Guids that should not be migrated because they have
+    # invalid/missing referents
+    short_missing_guids = MGuid.find(MQ('_id', 'in', short_missing) & MQ('is_orphaned', 'ne', True))
```

Comment:
Correct. So next time you migrate the prod db, take note of any `"Guid 123ab does not match..."` messages; they are not expected.